src/ggg/chained_context.rs

```diff
@@ -44,9 +44,18 @@ impl<'a> ChainedContextLookup<'a> {
             }
             2 => {
                 let coverage = Coverage::parse(s.read_at_offset16(data)?)?;
-                let backtrack_classes = ClassDefinition::parse(s.read_at_offset16(data)?)?;
-                let input_classes = ClassDefinition::parse(s.read_at_offset16(data)?)?;
-                let lookahead_classes = ClassDefinition::parse(s.read_at_offset16(data)?)?;
+                let backtrack_classes = match s.read::<Offset16>()?.to_usize() {
```

Comment:
Will `Option<Offset16>` look nicer?

Reply:
I don't get it, like this?

```rs
let backtrack_classes = match s.read::<Option<Offset16>>()?.to_usize() {
    Some(0) => ClassDefinition::Empty,
    Some(offset) => ClassDefinition::parse(data.get(offset..)?)?,
    None => ??
};
```

Reply:
Like this:

```rust
let backtrack_classes = match s.read::<Option<Offset16>>()? {
    Some(offset) => ClassDefinition::parse(data.get(offset.to_usize()..)?)?,
    None => ClassDefinition::Empty,
};
```

Reply:
Ah! I didn't know that `Option<Offset16>` has its own implementation for parse. 😄